### PR TITLE
[REVIEW] Add support for numpy RandomState handling in `sample`

### LINKED ISSUE #6127: [BUG] dask_cudf sample not working (TypeError)
**Describe the bug**
Using dask_cudf.sample raises a TypeError, but dask.dataframe.sample works just fine.

**Steps/Code to reproduce bug**
```python
import dask_cudf
import cudf
dask_cudf.from_cudf(cudf.DataFrame({"A": [0, 1, 2, 3, 4]}), npartitions=1).sample(frac=0.2).compute()
```

Output:
```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/base.py", line 167, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/base.py", line 452, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 527, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 471, in get_async
    fire_task()
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 466, in fire_task
    callback=queue.put,
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 516, in apply_sync
    res = func(*args, **kwds)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 227, in execute_task
    result = pack_exception(e, dumps)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/local.py", line 222, in execute_task
    result = _execute_task(task, data)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/dask/dataframe/methods.py", line 310, in sample
    return df.sample(random_state=rs, frac=frac, replace=replace) if len(df) > 0 else df
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/home/ssm-user/.conda/envs/ml/lib/python3.7/site-packages/cudf/core/frame.py", line 1757, in sample
    else np.int64(random_state)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'numpy.random.mtrand.RandomState'
```

**Expected behavior**
```python
import dask.dataframe as dd
dd.from_pandas(pd.DataFrame({"A": [0, 1, 2, 3, 4]}), npartitions=1).sample(frac=0.2).compute()
```

**Environment overview (please complete the following information)**
 - Environment location: AWS EC2 G4.2xlarge
 - Method of cuDF install: conda

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.4 LTS"
     NAME="Ubuntu"
     VERSION="18.04.4 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.4 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux ip-172-31-33-67 5.3.0-1033-aws #35-Ubuntu SMP Wed Aug 5 15:47:17 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

     ***GPU Information***
     Mon Aug 31 10:22:54 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Tesla T4            On   | 00000000:00:1E.0 Off |                    0 |
     | N/A   49C    P8    10W /  70W |      0MiB / 15109MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+

     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |  No running processes found                                                 |
     +-----------------------------------------------------------------------------+

     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              8
     On-line CPU(s) list: 0-7
     Thread(s) per core:  2
     Core(s) per socket:  4
     Socket(s):           1
     NUMA node(s):        1
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               85
     Model name:          Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
     Stepping:            7
     CPU MHz:             3201.865
     BogoMIPS:            4999.99
     Hypervisor vendor:   KVM
     Virtualization type: full
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            1024K
     L3 cache:            36608K
     NUMA node0 CPU(s):   0-7
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tscrep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke avx512_vnni

     ***CMake***
     /usr/local/bin/cmake
     cmake version 3.13.3

     CMake suite maintained and supported by Kitware (kitware.com/cmake).

     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

     ***nvcc***

     ***Python***
     /home/ssm-user/.conda/envs/ml/bin/python
     Python 3.7.8

     ***Environment Variables***
     PATH                            : /home/ssm-user/.conda/envs/ml/bin:/usr/local/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/ssm-user/.conda/envs/ml
     PYTHON_PATH                     :

     ***conda packages***
     /usr/local/condabin/conda
     # packages in environment at /home/ssm-user/.conda/envs/ml:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     _openmp_mutex             4.5                       1_gnu    conda-forge
     abseil-cpp                20200225.2           he1b5a44_2    conda-forge
     aiohttp                   3.6.2            py37h516909a_0    conda-forge
     aioitertools              0.7.0                      py_0    conda-forge
     arrow-cpp                 0.17.1          py37h1234567_11_cuda    conda-forge
     arrow-cpp-proc            1.0.1                      cuda    conda-forge
     async-timeout             3.0.1                   py_1000    conda-forge
     attrs                     20.1.0             pyh9f0ad1d_0    conda-forge
     aws-sdk-cpp               1.7.164              hba45d7a_2    conda-forge
     bokeh                     2.2.0            py37hc8dfbb8_0    conda-forge
     boost-cpp                 1.72.0               h7b93d67_2    conda-forge
     boto3                     1.14.51            pyh9f0ad1d_0    conda-forge
     botocore                  1.17.51            pyh9f0ad1d_0    conda-forge
     brotli                    1.0.7             he1b5a44_1004    conda-forge
     brotlipy                  0.7.0           py37h8f50634_1000    conda-forge
     bzip2                     1.0.8                h516909a_3    conda-forge
     c-ares                    1.16.1               h516909a_3    conda-forge
     ca-certificates           2020.6.20            hecda079_0    conda-forge
     certifi                   2020.6.20        py37hc8dfbb8_0    conda-forge
     cffi                      1.14.1           py37h2b28604_0    conda-forge
     chardet                   3.0.4           py37hc8dfbb8_1006    conda-forge
     click                     7.1.2              pyh9f0ad1d_0    conda-forge
     cloudpickle               1.6.0                      py_0    conda-forge
     cryptography              3.1              py37hb09aad4_0    conda-forge
     cudatoolkit               10.2.89              h6bb024c_0    nvidia
     cudf                      0.15.0          cuda_10.2_py37_g71cb8c0e0_0    rapidsai
     cudnn                     7.6.5                cuda10.2_0
     cuml                      0.15.0          cuda10.2_py37_ga3002e587_0    rapidsai
     cupy                      7.8.0            py37h940342b_0    conda-forge
     curl                      7.71.1               he644dc0_5    conda-forge
     cytoolz                   0.10.1           py37h516909a_0    conda-forge
     dask                      2.25.0                     py_0    conda-forge
     dask-core                 2.25.0                     py_0    conda-forge
     dask-cuda                 0.15.0                   py37_0    rapidsai
     dask-cudf                 0.15.0          py37_g71cb8c0e0_0    rapidsai
     distributed               2.25.0           py37hc8dfbb8_0    conda-forge
     dlpack                    0.3                  he1b5a44_1    conda-forge
     docutils                  0.15.2                   py37_0    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     faiss-proc                1.0.0                      cuda    conda-forge
     fastavro                  1.0.0.post1      py37h8f50634_0    conda-forge
     fastrlock                 0.5              py37h3340039_0    conda-forge
     freetype                  2.10.2               he06d7ca_0    conda-forge
     fsspec                    0.8.0                      py_0    conda-forge
     gflags                    2.2.2             he1b5a44_1004    conda-forge
     glog                      0.4.0                h49b9bf7_3    conda-forge
     gputil                    1.4.0              pyh9f0ad1d_0    conda-forge
     grpc-cpp                  1.30.2               heedbac9_0    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     icu                       67.1                 he1b5a44_0    conda-forge
     idna                      2.10               pyh9f0ad1d_0    conda-forge
     jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
     jmespath                  0.10.0             pyh9f0ad1d_0    conda-forge
     joblib                    0.16.0                     py_0    conda-forge
     jpeg                      9d                   h516909a_0    conda-forge
     krb5                      1.17.1               hfafb76e_2    conda-forge
     lcms2                     2.11                 hbd6801e_0    conda-forge
     ld_impl_linux-64          2.34                 hc38a660_9    conda-forge
     libblas                   3.8.0               17_openblas    conda-forge
     libcblas                  3.8.0               17_openblas    conda-forge
     libcudf                   0.15.0          cuda10.2_g71cb8c0e0_0    rapidsai
     libcuml                   0.15.0          cuda10.2_ga3002e587_0    rapidsai
     libcumlprims              0.15.0          cuda10.2_gdbd0d39_0    nvidia
     libcurl                   7.71.1               hcdd3856_5    conda-forge
     libedit                   3.1.20191231         he28a2e2_2    conda-forge
     libev                     4.33                 h516909a_0    conda-forge
     libevent                  2.1.10               hcdb4288_2    conda-forge
     libfaiss                  1.6.3           he61ee18_1_cuda    conda-forge
     libffi                    3.2.1             he1b5a44_1007    conda-forge
     libgcc-ng                 9.3.0               h24d8f2e_16    conda-forge
     libgfortran-ng            7.5.0               hdf63c60_16    conda-forge
     libgomp                   9.3.0               h24d8f2e_16    conda-forge
     libhwloc                  2.1.0                h3c4fd83_0    conda-forge
     libiconv                  1.16                 h516909a_0    conda-forge
     liblapack                 3.8.0               17_openblas    conda-forge
     libnghttp2                1.41.0               h8cfc5f6_2    conda-forge
     libopenblas               0.3.10          pthreads_hb3c22a3_4    conda-forge
     libpng                    1.6.37               hed695b0_2    conda-forge
     libprotobuf               3.12.4               h8b12597_0    conda-forge
     librmm                    0.15.0          cuda10.2_g8005ca5_0    rapidsai
     libssh2                   1.9.0                hab1572f_5    conda-forge
     libstdcxx-ng              9.3.0               hdf63c60_16    conda-forge
     libtiff                   4.1.0                hc7e4089_6    conda-forge
     libwebp-base              1.1.0                h516909a_3    conda-forge
     libxml2                   2.9.10               h68273f3_2    conda-forge
     llvmlite                  0.34.0           py37hf484d3e_0    numba
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.9.2                he1b5a44_3    conda-forge
     markupsafe                1.1.1            py37h8f50634_1    conda-forge
     msgpack-python            1.0.0            py37h99015e2_1    conda-forge
     multidict                 4.7.5            py37h8f50634_1    conda-forge
     nccl                      2.7.8.1              hc6a2c23_0    conda-forge
     ncurses                   6.2                  he1b5a44_1    conda-forge
     numba                     0.51.1          np1.11py3.7h04863e7_gc5f8505fb_0    numba
     numpy                     1.19.1           py37h7ea13bd_2    conda-forge
     olefile                   0.46                       py_0    conda-forge
     openssl                   1.1.1g               h516909a_1    conda-forge
     packaging                 20.4               pyh9f0ad1d_0    conda-forge
     pandas                    1.1.1            py37h3340039_0    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     pillow                    7.2.0            py37h718be6c_1    conda-forge
     pip                       20.2.2                     py_0    conda-forge
     psutil                    5.7.2            py37h8f50634_0    conda-forge
     pyarrow                   0.17.1          py37h1234567_11_cuda    conda-forge
     pycparser                 2.20               pyh9f0ad1d_2    conda-forge
     pynvml                    8.0.4                      py_1    conda-forge
     pyopenssl                 19.1.0                     py_1    conda-forge
     pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
     pysocks                   1.7.1            py37hc8dfbb8_1    conda-forge
     python                    3.7.8           h6f2ec95_1_cpython    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     python_abi                3.7                     1_cp37m    conda-forge
     pytz                      2020.1             pyh9f0ad1d_0    conda-forge
     pyyaml                    5.3.1            py37h8f50634_0    conda-forge
     re2                       2020.07.06           he1b5a44_1    conda-forge
     readline                  8.0                  he28a2e2_2    conda-forge
     rmm                       0.15.0          cuda_10.2_py37_g8005ca5_0    rapidsai
     s3fs                      0.4.2                      py_0    conda-forge
     s3transfer                0.3.3            py37hc8dfbb8_1    conda-forge
     scikit-learn              0.23.2           py37h6785257_0    conda-forge
     scipy                     1.5.2            py37hb14ef9d_0    conda-forge
     setuptools                50.0.0           py37hc8dfbb8_0    conda-forge
     six                       1.15.0             pyh9f0ad1d_0    conda-forge
     snappy                    1.1.8                he1b5a44_3    conda-forge
     sortedcontainers          2.2.2              pyh9f0ad1d_0    conda-forge
     spdlog                    1.7.0                hc9558a2_2    conda-forge
     sqlite                    3.33.0               h4cf870e_0    conda-forge
     tbb                       2019.9               hc9558a2_1    conda-forge
     tblib                     1.6.0                      py_0    conda-forge
     threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
     thrift-cpp                0.13.0               h62aa4f2_3    conda-forge
     tk                        8.6.10               hed695b0_0    conda-forge
     toolz                     0.10.0                     py_0    conda-forge
     tornado                   6.0.4            py37h8f50634_1    conda-forge
     treelite                  0.92             py37h023e13c_2    conda-forge
     treelite-runtime          0.92                     pypi_0    pypi
     typing_extensions         3.7.4.2                    py_0    conda-forge
     ucx                       1.8.1+g6b29558       cuda10.2_0    rapidsai
     ucx-py                    0.15.0+g6b29558          py37_0    rapidsai
     urllib3                   1.25.10                    py_0    conda-forge
     wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
     wrapt                     1.12.1           py37h8f50634_1    conda-forge
     xz                        5.2.5                h516909a_1    conda-forge
     yaml                      0.2.5                h516909a_0    conda-forge
     yarl                      1.3.0           py37h516909a_1000    conda-forge
     zict                      2.0.0                      py_0    conda-forge
     zlib                      1.2.11            h516909a_1009    conda-forge
     zstd                      1.4.5                h6597ccf_2    conda-forge

</pre></details>

**Additional context**
This could've easily found by doing good testing. Why was this bug not discovered?

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
closes #6127

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/frame.py [hunk 1]
```diff
@@ -1753,11 +1754,17 @@ def sample(
...
-                else np.int64(random_state)
-            )
+            if random_state is None:
+                seed = np.random.randint(
+                    np.iinfo(np.int64).max, dtype=np.int64
+                )
+            elif isinstance(random_state, np.random.mtrand.RandomState):
+                pos = random_state._bit_generator.state["state"]["pos"]
+                seed = random_state._bit_generator.state["state"]["key"][
+                    0 if pos == 624 else pos
```
Comment (galipremsagar) [inline python/cudf/cudf/core/frame.py:1764]: In the comments, Could we have some description of why we are using `624` here or mention this reference: https://stackoverflow.com/a/32172816 above?

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -1761,11 +1762,16 @@ def sample(
...
-                if random_state is None
-                else np.int64(random_state)
-            )
+            if random_state is None:
+                seed = np.random.randint(
+                    np.iinfo(np.int64).max, dtype=np.int64
+                )
+            elif isinstance(random_state, np.random.mtrand.RandomState):
+                pos = random_state._bit_generator.state["state"]["pos"]
+                keys = random_state._bit_generator.state["state"]["key"]
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py:1771]: Reaching into `_bit_generator` feels very fragile. Could we do anything that doesn't require reaching into internal attributes?

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/frame.py:1771]: We can use get_states, but it is just a list, so we might have to index it with integers

**Comment by rgsl888prabhu** [comment]
rerun tests

File: python/cudf/cudf/core/frame.py [hunk 3]
```diff
@@ -1759,8 +1759,8 @@ def sample(
                     np.iinfo(np.int64).max, dtype=np.int64
                 )
             elif isinstance(random_state, np.random.mtrand.RandomState):
-                pos = random_state._bit_generator.state["state"]["pos"]
-                keys = random_state._bit_generator.state["state"]["key"]
+                keys = random_state.get_state()[1]
+                pos = random_state.get_state()[2]
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py:1771]: How many items are in the list? Can we do something like:
```suggestion
                _, keys, pos = random_state.get_state()
```

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/frame.py:1771]: There are five items, and its a tuple. https://numpy.org/doc/stable/reference/random/generated/numpy.random.get_state.html#numpy.random.get_state

**Comment by rgsl888prabhu** [comment]
rerun tests

**Pull request merged**